frontend/controller/actions/group.js

```diff
@@ -185,26 +190,44 @@ export default (sbp('sbp/selectors/register', {
       // sync the group's contract state
       await sbp('chelonia/contract/sync', params.contractID)
 
-      // join the 'General' chatroom by default
-      const rootState = sbp('state/vuex/state')
-      const generalChatRoomId = rootState[params.contractID].generalChatRoomId
-      if (generalChatRoomId) {
-        await sbp('gi.actions/group/joinChatRoom', {
-          ...omit(params, ['options']),
-          data: {
-            chatRoomID: generalChatRoomId
-          },
-          hooks: {
-            prepublish: null,
-            postpublish: params.hooks?.postpublish
-          }
-        })
-      } else {
-        alert(L("Couldn't join the #{chatroomName} in the group. Doesn't exist.", { chatroomName: CHATROOM_GENERAL_NAME }))
-      }
-
       if (!params.options?.skipInviteAccept) {
+        // join the 'General' chatroom by default
+        const rootState = sbp('state/vuex/state')
+        const generalChatRoomId = rootState[params.contractID].generalChatRoomId
+        if (generalChatRoomId) {
+          await sbp('gi.actions/group/joinChatRoom', {
+            ...omit(params, ['options']),
+            data: {
+              chatRoomID: generalChatRoomId
+            },
+            hooks: {
+              prepublish: null,
+              postpublish: params.hooks?.postpublish
+            }
+          })
+        } else {
+          alert(L("Couldn't join the #{chatroomName} in the group. Doesn't exist.", { chatroomName: CHATROOM_GENERAL_NAME }))
+        }
+
         saveLoginState('joining', params.contractID)
+      } else {
+        /**
+         * Sync chatroom contracts he already joined
+         * if he tries to login in another device, he should skip to make any actions
+         * but he should sync all the contracts he was syncing in the previous device
+         */
+        const rootState = sbp('state/vuex/state')
+        const me = rootState.loggedIn.username
+        const chatRoomIds = Object.keys(rootState[params.contractID].chatRooms)
+          .filter(cId => rootState[params.contractID].chatRooms[cId].users.includes(me))
+
+        for (const cId of chatRoomIds) {
+          await sbp('chelonia/contract/sync', cId)
+        }
```

Comment:
Two minor change requests here, that you can make in your other PR after this is merged:

1. Since `const rootState = ...` is used in both if/else clauses, DRY it and move it to the top of the function so that it's written only once.
2. Chelonia is able to take an array of contractIDs, so it would be better to remove the `for` loop and simply pass in `chatRoomIds`